docs/recipes/add_dataset.md

```diff
@@ -0,0 +1,112 @@
+# Recipe: Adding a New Pretraining Dataset to Marin
+
+## Overview
+This recipe provides a structured guide for humans and AI agents to add new pretraining datasets to Marin. It emphasizes a step-by-step approach to ensure compatibility with Marin's data pipelines, such as tokenization and training workflows. Starting with schema inspection helps identify splits, subsets, and text fields, making the process more efficient and "roomba-friendly" (i.e., amenable to automation and junior contributors). These instructions also facilitate human onboarding by enforcing clear, reproducible practices.
+
+> **Reference:** Derived from community discussions on agent-friendly workflows, including @dlwh's checklist for dataset addition.
+
+## Prerequisites
+- Install the [Hugging Face `datasets` library](https://huggingface.co/docs/datasets/) (e.g., `pip install datasets`).
+- For YAML output, install `pyyaml` (e.g., `pip install pyyaml`).
+- Ensure access to the dataset: Hugging Face Hub ID, local path, or other supported formats.
+- Marin installed in editable mode (`pip install -e .`) for CLI access, or run the script standalone.
+
+## Step 1: Inspect Dataset Schema
+Inspect the dataset to determine its structure, including splits, subsets, and potential text fields. This ensures alignment with Marin's pretraining requirements, such as text-based tokenization (e.g., via `default_tokenize` in experiments).
+
+Use the `get_hf_dataset_schema` tool for automated analysis:
+
+### Command-Line Usage
+```sh
+python -m marin.tools.get_hf_dataset_schema <dataset-name-or-path> [options]
+```
+- **Examples:**
+  - Hugging Face dataset:
+    ```sh
+    python -m marin.tools.get_hf_dataset_schema roneneldan/TinyStories
+    ```
+  - Local path (e.g., directory or file):
+    ```sh
+    python -m marin.tools.get_hf_dataset_schema /path/to/local/dataset
+    ```
+  - Focus on a specific split:
+    ```sh
+    python -m marin.tools.get_hf_dataset_schema roneneldan/TinyStories --split train
+    ```
+  - Output as YAML:
+    ```sh
+    python -m marin.tools.get_hf_dataset_schema roneneldan/TinyStories --output yaml
+    ```
+
+### Import as a Function
+For programmatic use:
+```python
+from marin.tools.get_hf_dataset_schema import get_schema
+schema = get_schema("roneneldan/TinyStories", split="train")  # Optional: split, output_format
+print(schema)  # Or json.dumps(schema) for JSON
+```
+
+### Output Explanation
+The tool returns a dictionary (serialized as JSON or YAML) with:
+- `splits`: List of available splits (e.g., `["train", "validation"]`).
+- `subsets`: List of subsets, if applicable (often empty; e.g., language-specific variants).
```

Comment:
let's just skip this for now since we don't have it? Let's instead raise inside the tool if they are present